openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -394,7 +394,25 @@ def _on_user_authentication_failed(request):
         if is_request_from_mobile_app(request):
             return HttpResponse(status=401)
 
-        return redirect_to_login(request.path)
+        # .. toggle_name: REDIRECT_TO_LOGIN_ON_SAFE_SESSION_AUTH_FAILURE
+        # .. toggle_implementation: SettingToggle
+        # .. toggle_default: False
+        # .. toggle_description: Turn this toggle on if you want to redirect_to_login on every request
```

Comment:
Is this just a safety flag for rollout testing? If so, can you note that?

Reply:
Non blocking necessarily, but 'enable the flag to have it do what it used to do' is a weird pattern to me, especially if the flag defaults to false.

Reply:
Yep, this is a safety flag for rollout testing

Reply:
Nit: An updated message might be something like:
```
# .. toggle_description: Turn this toggle off to roll out new functionality, which returns a 401 rather than
   redirecting to login, when HTML is not expected by the client.
```